### PR TITLE
Add newsletter registration hook to email verification handler

### PR DESCRIPTION
**Pull request by samaij**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Currently the only way the newsletter registration hook (actionNewsletterRegistrationAfter) fires is when the module setting "Would you like to send a verification email after subscription?" is disabled. Add the hook to the email verification handler function, so it also fires when the setting is enabled and a customer verifies their registration.
| Type?         | bug fix
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | N/A
| How to test?  | Use [this test module](https://github.com/PrestaShop/ps_emailsubscription/files/7343826/newslettertest.zip). It logs to the PrestaShop log when the hook fires.

<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! --
>

### CODE REVIEW

**Review by PierreRambaud — APPROVED** [review]

**Comment by HanaRebaiQA** [comment]
Hello,

I tested this PR with 1.7.8.1 and 1.7.8.x with different tests cases.
The issue is fixed.
image

LGTM, QA :heavy_check_mark:

**Comment by kpodemski** [comment]
Thank you @samaij & @HanaRebaiQA

**Comment by kpodemski** [comment]
@matks @PierreRambaud I can't merge this PR 😱 

`The base branch restricts merging to authorized users.`

**Comment by PierreRambaud** [comment]
Thank you @samaij

**Comment by PierreRambaud** [comment]
> @matks @PierreRambaud I can't merge this PR scream
> 
> `The base branch restricts merging to authorized users.`

That's weird, you logically have maintainer access on this repository :thinking:

**Comment by devphoenix2404** [comment]
Hi Team,
Thanks for the fix 👍
I've a question, when I updated my shop with latest PS v1.7.8.2 version and ps_emailsubscription with v2.7.0 but I didn't find this code updated over there.
Can you guys please help me out to understand why the changes are not updated in the file, or is there anything I've missed.
Thanks

**Comment by tswfi** [comment]
@devphoenix2404 looks like this change is assigned to milestone 2.7.1 (https://github.com/PrestaShop/ps_emailsubscription/milestone/3?closed=1) which is not released yet.

**Comment by devphoenix2404** [comment]
Thanks @tswfi 👍 
Is there any tentative date? I need to update my client about it.

**Comment by tswfi** [comment]
@devphoenix2404 I don't know but I raised this for discussion in prestashop slack channel.

**Comment by devphoenix2404** [comment]
@tswfi Please keep posting and let me know so that I can update my client.

**Pull request merged**
